### PR TITLE
Home work for Lesson 3 - Main Data Classes

### PR DESCRIPTION
**Pull request by EugeniyPetrik**
1. Create a method which will take a natural number as an argument and will find a sum of all its digits
2. There is a string. Find a maximum number of digits placed in a row in it
3. There is a string. Find a maximal number (not a digit) in it
4. There are two strings. Find number of first chars from first string matching first chars of the second string. Consider two cases:
- strings are definitely different
- strings can completely match
5. There is an array of integers. First puts elements with even indexes and then with odd indexes
6. There is an array of integers (ary). Puts the index of the last element where ary[0]<ary[i]<ary[-1]
7. There is an array of integers (ary). Modify it with adding ary[0] (first element of the array) to each even number. Don't do it for first and last elements
8. There is a hash where keys and values are strings. Modify it: all keys should be symbols and all values should be integers. If you couldn't modify some value set it to nil
9. There is a hash where keys are symbols and values are integers. Modify it: remove all pairs where symbols start with "s" character
10. There is a hash where keys are symbols and values are integers. Modify it: leave only pairs where value is natural number and more than 0

### CODE REVIEW

File: Lesson_3_-_Main_Data_Classes/Tasks_data_&_classes.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+
+
+# 6. There is an array of integers (ary). Puts the index of the last element where ary[0]<ary[i]<ary[-1]
+
+puts "===== Task #6 =====" 
+
+arr = [45, 89, 123, 89, 46, 3, 9, 10, 98]
+
+puts "The index of the last element equals - #{arr.index(arr.last)}"
```
Comment (d-konovalov) [inline Lesson_3_-_Main_Data_Classes/Tasks_data_&_classes.rb]: the element should satisfy statement ary[0]<element<ary[-1]
so in your case we'll expect answer '4' (as 46 is more than 45 and less than 98)

File: Lesson_3_-_Main_Data_Classes/Tasks_data_&_classes.rb [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+puts "===== Task #8 =====" 
+
+hh8 = { 'name1' => 'John', 'last_name1' => 'Smith', 'age1' => '20', 'name2' => 'Matt', 'last_name2' => 'Walker', 'age2' => '25' }
+
+# puts hh8.each { |key, value| puts "#{key.to_sym} = #{value}" }
+
+# puts hh8.inject({}){ |memo,(key,value)| memo[key.to_sym] = value; memo }
+
+puts result = hh8.inject({}) { |result,(key, value)|
+    if value.match(/^\s*[+-]?((\d+_?)*\d+(\.(\d+_?)*\d+)?|\.(\d+_?)*\d+)(\s*|([eE][+-]?(\d+_?)*\d+)\s*)$/)
```
Comment (d-konovalov) [inline Lesson_3_-_Main_Data_Classes/Tasks_data_&_classes.rb]: what does this regex should catch?

Comment (EugeniyPetrik) [inline Lesson_3_-_Main_Data_Classes/Tasks_data_&_classes.rb]: This regex should catch values with integers
More compact recording is ' if value.match(/\d/) '

File: Lesson_3_-_Main_Data_Classes/Tasks_data_&_classes.rb (hunk 1 above)
Comment (EugeniyPetrik) [inline Lesson_3_-_Main_Data_Classes/Tasks_data_&_classes.rb]: last = arr.select {|i| i > arr[0] && i < arr[-1]}.last
arr.rindex(last)

**Pull request merged**